### PR TITLE
Use redis lists for save all job runtimes

### PR DESCRIPTION
**Pull request by davydovanton**
@mperham could you check this PR? It's simple realization for you comment https://github.com/davydovanton/sidekiq-statistic/pull/33#discussion_r34944402.

### CODE REVIEW

File: lib/sidekiq/statistic/middleware.rb [hunk 1]
```diff
@@ -30,25 +30,17 @@ def save_entry_for_worker(worker_status)
...
-            "#{worker_key}:min_time", min_time,
-            "#{worker_key}:max_time", max_time
+            "#{worker_key}:last_time", status[:last_runtime]
           ]
 
           redis.hincrby REDIS_HASH, "#{worker_key}:#{status[:last_job_status]}", 1
-          redis.hincrbyfloat REDIS_HASH, "#{worker_key}:total_time", status[:time]
 
           redis.hmset REDIS_HASH, statistics
+          redis.lpush "#{worker_key}:timeslist", status[:time]
```
Comment (mperham) [inline lib/sidekiq/statistic/middleware.rb]: You need to LTRIM too or else your list will continue to grow forever.

Comment (davydovanton) [inline lib/sidekiq/statistic/middleware.rb]: @mperham I want to save all job time in redis lists (list called `<yyyy-mm-dd>:<WorkerName>:timeslist`). Yes, for each day I have separate list for each worker. Do you think it's bad idea?

Comment (mperham) [inline lib/sidekiq/statistic/middleware.rb]: Some workers might be called millions of times in a day. That's a lot of data to save. Saving all data from the last hour seems ok.

Comment (davydovanton) [inline lib/sidekiq/statistic/middleware.rb]: @mperham yep, I understand. I think I can create special actor for calculation all time information.

**Comment by davydovanton** [comment]
So, @mperham, I changed logic for `timeslist` redis list. Now I remove all `timeslist` lists for each day after call `#statistic_hash` and calculate time information to `REDIS_HASH`. What do you think, it's better?

**Comment by mperham** [comment]
Yes, better.  Users might have other suggestions for how to optimize it further but that strategy will work for now.

**Comment by davydovanton** [comment]
Super, @mperham thanks for review! :+1:

**Pull request merged**